src/components/MultiSelect.tsx

```diff
@@ -0,0 +1,582 @@
+        fontFamily: 'Lato-Bold',
+        fontSize: 16,
+        flexWrap: 'wrap',
+        maxWidth: width - 150,
+        color: theme.textColor1
+      },
+      icon: {
+        fontSize: 26
+      },
+      rowView: {
+        flexDirection: 'row',
+        paddingLeft: 16
+      },
+      checkbox: {
+        flexDirection: 'column',
+        alignContent: 'center',
+        justifyContent: 'center'
+      },
+      iconView: {
+        width: 100,
+        alignItems: 'center',
+        justifyContent: 'center'
+      },
+      iconOnlyView: {
+        width: 200,
+        alignItems: 'center',
+        justifyContent: 'center'
+      },
+      iconOnly: {
+        paddingHorizontal: 16,
+        paddingVertical: 16,
+        fontSize: 60
+      }
+    });
+
+    return styles;
+  };
+
+  renderWithLabelAndIcon = (label: string, icon: string) => {
+    const { value, iconComponent, handleAdd, handleRemove } = this.props;
+
+    const theme = getTheme();
+    const styles = this.getStyles();
+
+    return (
+      <View style={styles.rowView}>
+        <View style={styles.checkbox}>
+          <CheckBox
+            value={this.props.checked}
+            onValueChange={() => {
+              if (!this.props.checked) {
+                handleAdd(value);
+              } else {
+                handleRemove(value);
+              }
+            }}
+          />
+        </View>
+        <View style={styles.iconView}>
+          {iconComponent ? (
+            React.createElement(iconComponent, {
+              name: icon,
+              style: styles.icon,
+              color: theme.textColor1
+            })
+          ) : (
+            <FontAwesome
+              name={icon}
+              color={theme.textColor1}
+              style={styles.icon}
+            />
+          )}
+        </View>
+        <View>
+          <Text style={styles.text}>{label}</Text>
+        </View>
+      </View>
+    );
+  };
+
+  renderWithLabel = (label: string) => {
+    const { value, handleAdd, handleRemove } = this.props;
+
+    const styles = this.getStyles();
+
+    return (
+      <View style={styles.rowView}>
+        <View style={styles.checkbox}>
+          <CheckBox
+            value={this.props.checked}
+            onValueChange={() => {
+              if (!this.props.checked) {
+                handleAdd(value);
+              } else {
+                handleRemove(value);
+              }
+            }}
+          />
+        </View>
+        <Text style={styles.text}>{label}</Text>
+      </View>
+    );
+  };
+
+  renderWithIcon = (icon: string) => {
+    const { value, iconComponent, handleAdd, handleRemove } = this.props;
+
+    const theme = getTheme();
+    const styles = this.getStyles();
+
+    return (
+      <View style={styles.rowView}>
+        <View style={styles.checkbox}>
+          <CheckBox
+            value={this.props.checked}
+            onValueChange={() => {
+              if (!this.props.checked) {
```

Comment:
Como tem o else, da para simplificar fazendo if (this.props.checked) ... remove else add, verificar outros lugares